### ISSUE TITLE
Objects should not be considered callable if `__call__` is an instance variable

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5796]
```python
class A:
    def __init__(self):
        self.__call__ = self.method1

    def method1(self, a: int):
        return a


# This should generate an error because `__call__` is
# callable only if it's a class variable.
A()(0)
```

This is related to [this issue](https://github.com/python/mypy/issues/2113) in the mypy issue tracker.

**Comment by erictraut** [comment]
This is included in pyright 1.1.325, which I just published. It will also be included in an upcoming pylance release.
